zrml/court/src/types.rs

```diff
@@ -0,0 +1,375 @@
+    pub cycle_ends: CycleEnds<BlockNumber>,
+    pub vote_item_type: VoteItemType,
+}
+
+/// The timing information about a court case.
+pub struct RoundTiming<BlockNumber> {
+    /// The end block of the pre-vote period.
+    pub pre_vote_end: BlockNumber,
+    /// The block duration for votes.
+    pub vote_period: BlockNumber,
+    /// The block duration for revealing votes.
+    pub aggregation_period: BlockNumber,
+    /// The block duration for appeals.
+    pub appeal_period: BlockNumber,
+}
+
+impl<BlockNumber: sp_runtime::traits::Saturating + Copy, Appeals: Default>
+    CourtInfo<BlockNumber, Appeals>
+{
+    pub fn new(round_timing: RoundTiming<BlockNumber>, vote_item_type: VoteItemType) -> Self {
+        let pre_vote = round_timing.pre_vote_end;
+        let vote = pre_vote.saturating_add(round_timing.vote_period);
+        let aggregation = vote.saturating_add(round_timing.aggregation_period);
+        let appeal = aggregation.saturating_add(round_timing.appeal_period);
+        let cycle_ends = CycleEnds { pre_vote, vote, aggregation, appeal };
+        let status = CourtStatus::Open;
+        Self { status, appeals: Default::default(), cycle_ends, vote_item_type }
+    }
+
+    pub fn update_lifecycle(&mut self, round_timing: RoundTiming<BlockNumber>) {
+        self.cycle_ends.pre_vote = round_timing.pre_vote_end;
+        self.cycle_ends.vote = self.cycle_ends.pre_vote.saturating_add(round_timing.vote_period);
+        self.cycle_ends.aggregation =
+            self.cycle_ends.vote.saturating_add(round_timing.aggregation_period);
+        self.cycle_ends.appeal =
+            self.cycle_ends.aggregation.saturating_add(round_timing.appeal_period);
+    }
+}
+
+/// After a juror was randomly selected to vote in a court case,
+/// this information is relevant to handle the post-selection process.
+#[derive(
+    parity_scale_codec::Decode,
+    parity_scale_codec::Encode,
+    parity_scale_codec::MaxEncodedLen,
+    scale_info::TypeInfo,
+    Clone,
+    Debug,
+    PartialEq,
+    Eq,
+)]
+pub struct Draw<AccountId, Balance, Hash, DelegatedStakes> {
+    /// The juror who was randomly selected.
+    pub juror: AccountId,
+    /// The weight of the juror in this court case.
+    /// The higher the weight the more voice the juror has in the final winner decision.
+    pub weight: u32,
+    /// The information about the vote state.
+    pub vote: Vote<Hash, DelegatedStakes>,
+    /// The amount of funds which can be slashed for this court case.
+    /// This is related to a multiple of `MinStake` to mitigate Sybil attacks.
+    pub slashable: Balance,
+}
+
+/// All information related to one item in the stake weighted juror pool.
+#[derive(
+    parity_scale_codec::Decode,
+    parity_scale_codec::Encode,
+    parity_scale_codec::MaxEncodedLen,
+    scale_info::TypeInfo,
+    Clone,
+    Debug,
+    PartialEq,
+    Eq,
+)]
+pub struct JurorPoolItem<AccountId, Balance> {
```

Comment:
`unconsumed_stake = stake - consumed_stake`

`stake = unconsumed_stake + consumed_stake` 

`stake` is the overall amount which was put in by the juror or delegator. It is only decreased, when the juror or delegator decides to exit the court. Otherwise this `stake` amount serves as the key inside the court pool to search for the pool item. So it shouldn't be ever changed, unless you want to modify the juror pool with it too.